freud/diffraction.pyx

```diff
@@ -617,3 +884,31 @@ cdef class DiffractionPattern(_Compute):
             return freud.plot._ax_to_bytes(self.plot())
         except (AttributeError, ImportError):
             return None
+
+def reciprocal_isotropic(box, k_max, k_min=0, max_k_points=10000):
```

Comment:
I created this method just so that I could test and optimize this function directly from Python. @DomFijan Do you think there is any need to expose this in the public API? If not, I'd like to remove it from the Python API and just leave it in C++.

Reply:
I agree. No need to expose it in python API. I would advocate to move some of the documentation information to the StaticStructurefactorDirect class such as the similarity with dynasor implementation and at what max_k_points they start to give same results.

Reply:
Done. There was one change that I wanted to mention:

> and at what max_k_points they start to give same results.

The information that was in the docstring was incorrect. It said "This implementation approaches Dynasor implementation for number of max_k_points around 50000, but is also faster." I don't know where that came from. [By default, Dynasor uses 20,000 points.](https://gitlab.com/materials-modeling/dynasor/-/blob/1.1/dynasor#L146-147) The scaling should be exactly the same since we're using the same probability function for each k vector to ensure that reciprocal space is sampled isotropically. However, this also depends on the value of `k_max`, because for smaller values of `k_max` we can sample the entire grid in reciprocal space up to the target magnitude of k. That's met by [this condition for skipping the pruning step](https://github.com/glotzerlab/freud/blob/aba73322c379a232aed8ea9b69fb4ebb4b4d376d/cpp/diffraction/StaticStructureFactorDirect.cc#L186-L190).

Reply:
I was under the impression that the two implementations are different, and I went to test this at some point and was getting different results for lower numbers of max_k_points (or at least I thought I did this). I did a couple of tests on current build and it seems that this is no longer the case. Everything seems to be in line with what you have written...